src/foremast/s3/s3deploy.py

```diff
@@ -107,12 +107,55 @@ def _upload_artifacts_to_version(self):
         """Recursively upload directory contents to S3."""
         if not os.listdir(self.artifact_path) or not self.artifact_path:
             raise S3ArtifactNotFound
-        cmd = 'aws s3 sync {} {} --delete --exact-timestamps --profile {}'.format(self.artifact_path,
-                                                                                  self.s3_version_uri, self.env)
-        result = subprocess.run(cmd, check=True, shell=True, stdout=subprocess.PIPE)
-        LOG.debug("Upload Command Ouput: %s", result.stdout)
+
+        uploaded = False
+        if self.s3props.get("content_encodings"):
+            LOG.info("Uploading in multiple parts to set content encoding")
+            uploaded = self.content_encoding_uploads()
+
+        if not uploaded:
+            cmd = 'aws s3 sync {} {} --delete --exact-timestamps --profile {}'.format(self.artifact_path,
+                                                                                      self.s3_version_uri, self.env)
+            result = subprocess.run(cmd, check=True, shell=True, stdout=subprocess.PIPE)
+            LOG.debug("Upload Command Ouput: %s", result.stdout)
+
         LOG.info("Uploaded artifacts to %s bucket", self.bucket)
 
+    def content_encoding_uploads(self):
+        """Finds all specified encoded directories and uploads in multiple parts,
+        setting content-encoding for compressed dirs.
+
+        Returns:
+            bool: True if uploaded
+        """
+        excludes_str = ''
+        includes_cmds = []
+        cmd_base = 'aws s3 sync {} {} --delete --exact-timestamps --profile {}'.format(self.artifact_path,
+                                                                                       self.s3_version_uri,
+                                                                                       self.env)
+
+        for content in self.s3props.get('content_encodings'):
+            full_path = os.path.join(self.artifact_path, content['path'])
+            if not os.listdir(full_path):
+                raise S3ArtifactNotFound
+
+            excludes_str += '--exclude "{}/*" '.format(content['path'])
+            include_cmd = '{} --exclude "*", --include "{}/*"'.format(cmd_base, content['path'])
+            include_cmd += ' --content-encoding {} --metadata-directive REPLACE'.format(content['encoding'])
```

Comment:
Can we change the field to be the header name to reduce confusion?
`encoding` -> `content-encoding`